python/adelphi/adelphi/anonymize.py

```diff
@@ -38,59 +33,70 @@ def get_name(original_name, prefix):
-                       for field_type in udt.field_types]
 
+def anonymize_udts(udts):
+    # anonymize all udt names first
+    for udt in udts:
+        udt.keyspace = get_name(udt.keyspace, KEYSPACE_PREFIX)
+        udt.name = get_name(udt.name, TYPE_PREFIX)
+    
+    for udt in udts:
+        # field names
+        udt.field_names = [get_name(field_name, FIELD_PREFIX) for field_name in udt.field_names]
+        # field types
+        udt.field_types = [anonymize_type(field_type) for field_type in udt.field_types]
 
-def anonymize_column(column, prefix):
-    column.name = get_name(column.name, prefix)
+def anonymize_column(column):
+    column.name = get_name(column.name, COLUMN_PREFIX)
+    column.cql_type = anonymize_type(column.cql_type)
 
 
 def anonymize_index(index):
     index.name = get_name(index.name, INDEX_PREFIX)
-    prefix = COLUMN_PREFIX if index.index_options['target'] in name_map[COLUMN_PREFIX] \
-        else CLUSTERING_KEY_PREFIX
-    index.index_options['target'] = name_map[prefix][index.index_options["target"]]
-    index.keyspace_name = name_map[KEYSPACE_PREFIX][index.keyspace_name]
-    index.table_name = name_map[TABLE_PREFIX][index.table_name]
-
+    index.index_options['target'] = get_name(index.index_options["target"], COLUMN_PREFIX)
+    index.keyspace_name = get_name(index.keyspace_name, KEYSPACE_PREFIX)
+    index.table_name = get_name(index.table_name, TABLE_PREFIX)
 
 def anonymize_table(table):
     table.keyspace_name = get_name(table.keyspace_name, KEYSPACE_PREFIX)
     table.name = get_name(table.name, TABLE_PREFIX)
 
-    for partition_key in table.partition_key:
-        anonymize_column(partition_key, PARTITION_KEY_PREFIX)
+    for pk in table.partition_key:
+        anonymize_column(pk)
 
-    for clustering_key in table.clustering_key:
-        anonymize_column(clustering_key, CLUSTERING_KEY_PREFIX)
+    for ck in table.clustering_key:
+        anonymize_column(ck)
 
-    # CK are also in the standard columns, but different objects
-    # if we don't anonymize them there too, the generated cql is wrong
-    for clustering_key in [t for t in table.columns.values() if t.name in name_map[CLUSTERING_KEY_PREFIX]]:
-        clustering_key.name = name_map[CLUSTERING_KEY_PREFIX][clustering_key.name]
+    # remove comment since it may contain sensitive information
+    table.options["comment"] = ""
 
+    # The CQL python driver holds the same object references for clustering keys
+    # and regular columns, so we have to separate them otherwise the names
+    # will be re-anonymized.
     for column in get_standard_columns_from_table_metadata(table):
-        anonymize_column(column, COLUMN_PREFIX)
+        anonymize_column(column)
 
     for index in list(table.indexes.values()):
-        if (index.index_options["target"] not in name_map[COLUMN_PREFIX].keys() and
-                index.index_options["target"] not in name_map[CLUSTERING_KEY_PREFIX].keys()):
+        if (index.kind == "CUSTOM"):
```

Comment:
We want to skip `CUSTOM` indices because we cannot recreate them without the custom Java class with the implementation. The name of the class could potentially contain sensitive information too. We can probably go a step further in a future PR to only remove custom indexes that do not contain a C* indexing class.

Reply:
Yeah, I was gonna mention the same thing... _some_ indexes we could re-create.  But I'm not sure it's a huge problem... waiting until a future release doesn't seem unreasonable.  We'll lose _some_ information about the use of these "custom" indices for the time being but that seems like a lower priority concern than getting the schemas in the first place.

Reply:
Perhaps create a ticket for this future work now, though?

Reply:
FTR I've created https://github.com/datastax/adelphi/issues/61 to track this future work